lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -121,12 +112,16 @@ def validate_config_files
 
     private
 
+    def kubeclient_configs
+      @kubeclient_configs ||= @kubeconfig_files.map { |f| KubeConfig.read(f) if File.file?(f) }.compact
```

Comment:
It might be worth rejecting kubeconfig entries that aren't files in `initialize` so that we don't need to make the check multiple times (here and in `validate_config_files`)

Reply:
If we reject them up front though, any related errors we show will not match the list of files the user thinks they're giving us, which could be confusing. E.g. notice how the error I added below uses `@kubeconfig_files` directly (which is unfiltered). Putting the validation in the initializer would be one way around that problem, but I have found that making initializers raise (other than for malformed input) is unexpected and makes it more complicated for consumers to handle the errors as intended.

Reply:
Actually, maybe we should just run the whole validation in `build_kubeclient` instead of proceeding as long as the files we can load contain the relevant context. Since contexts are merged across files, it is possible that we have a context with the correct name that is not the one the user expected to use.

Reply:
>but I have found that making initializers raise (other than for malformed input) is unexpected and makes it more complicated for consumers to handle the errors as intended.

Fair point, I had the same idea in the back of my mind when I commented but decided to ask nonetheless.

🚢